Simplenote/src/main/java/com/automattic/simplenote/NoteEditorFragment.java

```diff
@@ -440,6 +443,20 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
         mTagInput.setTokenizer(new SpaceTokenizer());
         mTagInput.setAdapter(mTagAutocompleteAdapter);
         mTagInput.setOnFocusChangeListener(this);
+        mTagInput.setOnKeyListener(new OnKeyListener() {
+            @Override
+            public boolean onKey(View view, int keyCode, KeyEvent event) {
+                if (!mIsMarkdownEnabled) {
+                    EditText editText = (EditText) view;
+                    if (editText.getSelectionStart() == editText.getText().length()) {
+                        if (keyCode == KeyEvent.KEYCODE_DPAD_RIGHT) {
+                            return true;
+                        }
+                    }
+                }
+                return false;
+            }
+        });
```

Comment:
```suggestion
        mTagInput.setOnKeyListener(new OnKeyListener() {
            @Override
            public boolean onKey(View view, int keyCode, KeyEvent event) {
                return !mIsMarkdownEnabled && mTagInput.getSelectionStart() == mTagInput.getText().length() && keyCode == KeyEvent.KEYCODE_DPAD_RIGHT;
            }
        });
```

Can we use the `mTagInput` variable instead of creating the `editText` variable from the `view` itself?  If we remove the `editText` variable, we can simplify this cascade of `if` statements into a single `if` statement like this.

```java
if (!mIsMarkdownEnabled && mTagInput.getSelectionStart() == mTagInput.getText().length() && keyCode == KeyEvent.KEYCODE_DPAD_RIGHT) {
    return true;
}
```

Since that `if` statement is just returning `true` if true and `false` otherwise, we can simplify the entire block to something like this.

```java
mTagInput.setOnKeyListener(new OnKeyListener() {
    @Override
    public boolean onKey(View view, int keyCode, KeyEvent event) {
        return !mIsMarkdownEnabled && mTagInput.getSelectionStart() == mTagInput.getText().length() && keyCode == KeyEvent.KEYCODE_DPAD_RIGHT;
    }
});
```

Reply:
Great suggestion @theck13, many thanks, indeed the code gets much more simlified! 🌟 🤗 

This is now done, you can check the change here: 8714b3881874cfb9f5f6735eb89b5f8b84c7df3f